packages/job-worker/src/studio/model/StudioBaselineHelper.ts

```diff
@@ -49,9 +61,62 @@ export class StudioBaselineHelper {
 						this.#pendingExpectedPackages
 				  )
 				: undefined,
+			this.#routeSetChanged
+				? this.#context.directCollections.Studios.update(
+						{
+							_id: this.#context.studioId,
+						},
+						{
+							$set: { 'routeSets.overrides': this.#overridesRouteSetBuffer.overrides },
+						}
+				  )
+				: undefined,
 		])
 
 		this.#pendingExpectedPlayoutItems = undefined
 		this.#pendingExpectedPackages = undefined
+		this.#routeSetChanged = false
+	}
+
+	updateRouteSetActive(routeSetId: string, isActive: boolean): void {
+		const studio = this.#context.studio
+		const saveOverrides = (newOps: SomeObjectOverrideOp[]) => {
+			this.#overridesRouteSetBuffer = { defaults: this.#overridesRouteSetBuffer.defaults, overrides: newOps }
+			this.#routeSetChanged = true
+		}
+		const overrideHelper = useOverrideOpHelperBackend(saveOverrides, this.#overridesRouteSetBuffer)
+
+		const routeSets: WrappedOverridableItemNormal<StudioRouteSet>[] = getAllCurrentItemsFromOverrides(
+			this.#overridesRouteSetBuffer,
+			null
+		)
+
+		const routeSet = Object.values<WrappedOverridableItemNormal<StudioRouteSet>>(routeSets).find((routeSet) => {
+			return routeSet.id === routeSetId
+		})
+
+		if (routeSet === undefined) throw new Error(`RouteSet "${routeSetId}" not found!`)
+		if (routeSet.computed?.behavior === StudioRouteBehavior.ACTIVATE_ONLY && isActive === false)
+			throw new Error(`RouteSet "${routeSet.id}" is ACTIVATE_ONLY`)
+
+		logger.debug(`switchRouteSet "${studio._id}" "${routeSet.id}"=${isActive}`)
+		overrideHelper.setItemValue(routeSet.id, `active`, isActive)
```

Comment:
Yes or maybe change just back to `activeRouteSets: Record<string, boolean>` and then only do the recalculation of the override changes next time the saveAlleToDatabase() is called. Instead of saving a whole recalculation for every time an active state is changed in updateRouteSetActive()
I my opinion another place in Mongo just for activeRouteSets is a bit of a workaround, cause then we would need to do that every time there was a new parameter, that had to be changed form the API.

Reply:
The main reason for moving overridesOpHelper to lib was to use the functionality of the OverrideOpHelperImpl class to update the overrides Array. But are there another feature set for the backend?
(like the applyAndValidateOverrides are there for getting the calculated routeSet with overides)

Reply:
> then only do the recalculation of the override changes next time the saveAlleToDatabase() is called.

Care would need to be taken here, because the order of operations matters. If routeset A and B are in an exclusive group, only the last call for that group should be applied.

> I my opinion another place in Mongo just for activeRouteSets is a bit of a workaround, cause then we would need to do that every time there was a new parameter, that had to be changed form the API.

Whether this is possible or not depends on 'Do the blueprints need to be able to specify a default for active?'

Modifying an `ObjectWithOverrides` during playout is not a use case that was considered, this flow is used solely for 'configuration', where the  `defaults` are provided by blueprints and `overrides` defined by a user.
So arguably, the `active` property does not fall into that 'configuration' category as it is modified during normal playout, so should it be part of the structure, or should there be another place to define the playout properties of routesets?

I'm not certain that a new property on the studio is the best solution, but it is an option.

> I my opinion another place in Mongo just for activeRouteSets is a bit of a workaround, cause then we would need to do that every time there was a new parameter, that had to be changed form the API.

Yes, but in reality is any other part of the routeset something that would want to be changed as part of playout? If it is not and is configuration, then it would remain on the current object.

-----

> But are there another feature set for the backend?

No, because the `overrides` is all user owned data, the backend is not supposed to be modifying it. Except for in a couple of migrations. 
Of course, as part of a shift away from meteor that would probably change, moving the OverrideOpHelper into the backend so that the ui can use simple rest calls to make its changes.